core/lib/src/response/named_file.rs

```diff
@@ -78,6 +78,31 @@ impl NamedFile {
     pub fn path(&self) -> &Path {
         self.0.as_path()
     }
+
+    /// Sets the default content type for the file
+    ///
+    /// If the file does not have an extension, the content type will
+    /// be set to this value. If no default is set and the file does not
+    /// have an extension, no content-type header will be sent.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use std::io;
+    /// use rocket::response::NamedFile;
+    /// use rocket::http::ContentType;
+    ///
+    /// # #[allow(head_code)]
+    /// # async fn demo_default_type() -> io::Result<()> {
+    /// let mut file = NamedFile::open("foo.text").await?;
+    /// file.default_type(ContentType::Plain);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn default_type(&mut self, content_type: ContentType) -> &mut Self {
```

Comment:
I would rather this be a builder method, taking `self` by value.
```suggestion
    pub fn with_default(mut self, content_type: ContentType) -> Self {
```

Then the example can be:

```rust
#[get("/<path..>")]
fn file(path: PathBuf) -> Result<NamedFile, Debug<io::Error>> {
    Ok(NamedFile::open(path).await?.with_default(ContentType::Plain))
}
```

Which I think is much more illustrative.